### ISSUE TITLE
__docval__['arg'] for links not preserving type

### ISSUE THREAD

**Issue by luiztauffer** [hdmf-dev/hdmf#475]
In the `ndx-fret` extension, we’re defining a link to a Device 
[spec code](https://github.com/catalystneuro/ndx-fret/blob/master/src/spec/create_extension_spec.py)

```python
FRETSeries = NWBGroupSpec(
    neurodata_type_def='FRETSeries',
    neurodata_type_inc='ImageSeries',
    doc='Donor/Acceptor specific information.',
)

FRETSeries.add_link(
    name='device',
    doc='The device that was used to record.',
    target_type='Device',
)
```

but when importing FRETSeries and checking its args:
```python
FRETSeries.__init__.__docval__['args'][3]
```
I get:
```
{'name': 'device',
'doc': 'The device that was used to record.',
'type': hdmf.container.AbstractContainer}
```
when I was hoping to get:
```
{'name': 'device',
 'doc': 'The device that was used to record.',
 'type': pynwb.device.Device}
```

Other classes that link to Devices, e.g. ElectrodeGroup, have the docval arg type correctly set as Device. Why this is not happening on the extension? What is the best way to solve this issue?
Right now I’m solving this by manually setting the arg type to Device at the [module init](https://github.com/catalystneuro/ndx-fret/blob/master/src/pynwb/ndx_fret/__init__.py)

The arg type as hdmf.container.AbstractContainer works, we can use the extension. But we are using the docval args to make json schemas and gui forms automatically from pynwb classes, and for that we need the link type to be properly set as Device.
The issue seems to be happening only for links, added attributes and groups are preserving the arg types.

**Comment by rly** [comment]
@luiztauffer Thanks for the bug report. I'm not sure why docval types for links were always set to `AbstractContainer` but that is changed now and will be in the next release.

**Comment by luiztauffer** [comment]
@rly related to this, with the latest release for `hdmf==2.3.0` and pre-release `pynwb==1.4.0.post0.dev6` 
The docval type for the link field is being preserved, but the class method `.get_fields_conf()` returns the field as a child:
```
{'name': 'device',
  'child': True,
  'doc': 'The device that was used to record.'}
```

But this field is defined as a link, I expected this dictionary to not have a key 'child'

**Comment by rly** [comment]
@luiztauffer this should be fixed now in the dev branch after #535 and will be released later this week. Sorry for missing this earlier.
